File: bruteforce/b1816/Solution.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int loop = Integer.parseInt(br.readLine().strip());

        for (int i = 0; i < loop; i++) {
            Long n = Long.parseLong(br.readLine().strip());

            for (int j = 2; j < 1000001; j++) {
                if (n % j == 0) {
                    System.out.println("NO");
                    break;
                }

                if (j == 1000000) {
                    System.out.println("YES");
                }
            }
        }
        br.close();
    }

}

File: bruteforce/b1816/solution.py
import sys

input = sys.stdin.readline

n = int(input())

for _ in range(n):
    s = int(input())
    for i in range(2, 1_000_001):
        if s % i == 0:  # 100만 이하의 약수(소인수 포함)가 있다는 뜻
            print("NO")
            break
    else:
        print("YES")
